src/api/base.ts

```diff
@@ -0,0 +1,59 @@
+import ky from 'ky';
+
+import { ERROR_MESSAGE } from '../utils/constants';
+import { ApiError } from '../types';
+
+// const apiUrl = process.env.NEXT_PUBLIC_API_URL;
+const apiUrl = 'https://sportujspolu-api.onrender.com/api/v1/';
+
+export const api = ky.create({
+  prefixUrl: apiUrl,
+  headers: {},
+});
+
+export const privateApi = (token: string) =>
+  ky.create({
+    prefixUrl: apiUrl,
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+export const apiPost = async <T>(
+  endpoint: string,
+  body: unknown,
+  token?: string,
+  onError?: (error: ApiError) => void,
+): Promise<T> => {
```

Comment:
Do we really need this implicit return?

Reply:
Oh. Is it needed for the generic?

Reply:
`return await apiInstance.post(endpoint, { json: body }).json<T>();` would  do the same it seems. No explicit return is needed